src/NServiceBus.AcceptanceTests/Core/Routing/When_routing_interface_message.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.Routing
+{
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using AcceptanceTesting.Customization;
+    using NUnit.Framework;
+
+    public class When_routing_interface_message : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_route_correctly()
+        {
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(c => c.When(b => b.SendLocal(new StartMessage())))
+                .Done(c => c.GotTheMessage)
+                .Run();
+
+            Assert.True(context.GotTheMessage);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool GotTheMessage { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>((c, r) =>
+                {
+                    var routing = c.ConfigureTransport().Routing();
+                    routing.RouteToEndpoint(typeof(IMyMessage), typeof(Endpoint));
+                });
+            }
+
+            public class StartMessageHandler : IHandleMessages<StartMessage>
+            {
+                public IMessageCreator MessageCreator { get; set; }
+
+                public Task Handle(StartMessage message, IMessageHandlerContext context)
+                {
+                    var interfaceMessage = MessageCreator.CreateInstance<IMyMessage>();
+                    return context.Send(interfaceMessage);
```

Comment:
couldn't we just do that in the `When` delegate? (I guess resolving the `IMessageCreator` could be an issue?)

Reply:
Yea, couldn’t find a way to access the IBuilder there tho. Ideas?